distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/platform/TestSingleSignOn.java

```diff
@@ -87,182 +113,249 @@ public void beforeTest() throws Exception {
+        // Start the services needed for testing.
+        // We need to start the Search UI to test that it redirects properly
         getServiceManager().startFeature(true, "security-idp", "search-ui-app");
 
-        // Create a validator from the metadata schema so that we can validate the metadata
-        SchemaFactory schemaFactory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
-        Schema schema = schemaFactory.newSchema(getClass().getClassLoader().getResource("saml-schema-metadata-2.0.xsd"));
-        Validator validator = schema.newValidator();
-
-        // Get the metadata
-        String serverMetadata = get(SERVICE_ROOT + "/idp/login/metadata").asString();
-        String clientMetadata = get(SERVICE_ROOT + "/saml/sso/metadata").asString();
-
-        // Ensure the metadata is valid according to the SAML Metadata schema
-        validator.validate(new StreamSource(new StringReader(clientMetadata)));
-        validator.validate(new StreamSource(new StringReader(serverMetadata)));
-
-        // To find the right inputs for the settings, go into the metatype.xml file.
-        // The key is the "id" and the value type is determined by the cardinality as such:
-        // Positive = array, negative = vector, 0 (none) = single variable
-        Dictionary<String, Object> clientSettings = new Hashtable<>();
-        clientSettings.put("metadata", serverMetadata);
-        Dictionary<String, Object> serverSettings = new Hashtable<>();
-        serverSettings.put("spMetadata", new String[] {clientMetadata});
-
-        // Update the client and server with the metadata
-        Configuration clientConfig = getAdminConfig()
-                .getConfiguration("org.codice.ddf.security.idp.client.IdpMetadata", null);
-        Configuration serverConfig = getAdminConfig()
-                .getConfiguration("org.codice.ddf.security.idp.server.IdpEndpoint", null);
-        clientConfig.update(clientSettings);
-        serverConfig.update(serverSettings);
-        expect("Configs to update").within(2, TimeUnit.MINUTES)
-                .until(clientConfig::getProperties, notNullValue());
-        expect("Configs to update").within(2, TimeUnit.MINUTES)
-                .until(serverConfig::getProperties, notNullValue());
+        // Get all of the metadata
+        String metadata = get(SERVICE_ROOT + "/idp/login/metadata").asString();
+        String ddfSpMetadata = get(SERVICE_ROOT + "/saml/sso/metadata").asString();
+
+        // Make sure all the metadata is valid before we set it
+        validateSaml(metadata, SamlSchema.METADATA);
+        validateSaml(ddfSpMetadata, SamlSchema.METADATA);
+
+        // Set the metadata
+        setConfig("org.codice.ddf.security.idp.client.IdpMetadata", "metadata", metadata);
+        setConfig("org.codice.ddf.security.idp.server.IdpEndpoint", "spMetadata", new String[] {ddfSpMetadata});
```

Comment:
That's a very good question! The IdP Endpoint can have more than one set of service provider metadata, so when you set the configuration it expects an array. Therefore, even though it is a single item, I must wrap it in an array when I set the configuration. 

If you look at the metatype.xml file for the IdP Server [click here](https://github.com/spearskw/ddf/blob/1841db3e2b1f2f34159b73859cba19359d4fae54/platform/security/idp/security-idp-server/src/main/resources/OSGI-INF/metatype/metatype.xml), you will notice that the attribute with id="spMetadata" has a cardinality="100", while the metatype.xml for the IdP Client [click here](https://github.com/spearskw/ddf/blob/1b2b8e8596109d7aca970c1da171221460ea9420/platform/security/idp/security-idp-client/src/main/resources/OSGI-INF/metatype/metatype.xml) has an attribute with id="metadata" and no cardinality. Positive cardinality in the metatype.xml means it's an array, negative means it's a vector, and nothing means it's a single value.

Definitely not obvious. Should I add a comment to the java outlining what I said here?